src/main/java/org/sagebionetworks/web/client/widget/entity/menu/v3/DefaultEntityActionMenuLayoutUtil.java

```diff
@@ -0,0 +1,532 @@
+package org.sagebionetworks.web.client.widget.entity.menu.v3;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.sagebionetworks.repo.model.EntityType;
+import org.sagebionetworks.repo.model.EntityTypeUtils;
+import org.sagebionetworks.web.client.jsinterop.ActionViewProps;
+import org.sagebionetworks.web.client.jsinterop.SxProps;
+import org.sagebionetworks.web.client.jsinterop.entity.actionmenu.EntityActionMenuLayout;
+import org.sagebionetworks.web.client.place.Synapse;
+import org.sagebionetworks.web.client.widget.entity.menu.v2.Action;
+
+public class DefaultEntityActionMenuLayoutUtil {
```

Comment:
More configuration-as-code. If this were all in TS/React, this would probably be nicer to have in JSON, but it's very SWC specific so it's left as Java code. If there's a better way to do this, let me know.

Reply:
That's what I was thinking as well, but this verbose java looks right to me.